libs/net/dal/Extensions/ContentExtensions.cs

```diff
@@ -171,14 +173,23 @@ public static TNOContext UpdateContext(this TNOContext context, Content original
         {
             var current = a.TonePoolId != 0 ? oTonePools.FirstOrDefault(o => o.TonePoolId == a.TonePoolId) : null;
             if (current == null)
-                original.TonePoolsManyToMany.Add(a);
+            {
+                a.TonePool ??= context.TonePools.FirstOrDefault(tp => tp.Id == a.TonePoolId);
+                tonePoolsToAdd.Add(a);
```

Comment:
Why create a separate list that you then iterate over again?  Just do the logic here.